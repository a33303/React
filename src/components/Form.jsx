import React, { useState } from "react"

export function Form ({ addMessage }) {
    const [text, setText] = useState('')

    const addText = (event) => {
        event.preventDefault()
        addMessage({text, author:'User'})
        setText('')
    }

    return (
        <form onSubmit={addText}>
            <input 
            type="text"
            value={text} 
            onChange={event => setText(event.target.value)}/>
            <button type="submit">Send</button>
        </form>
    )
}
