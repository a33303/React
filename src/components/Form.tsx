import React, { useState, FC, useContext } from 'react';
import { Input, Button } from '@mui/material';
import { ThemeContext } from '../utils/ThemeContext';
interface Message {
  text: string;
  author: string;
}

interface FormProps {
  addMessage: (message: Message) => void;
}

export const Form: FC<FormProps> = ({ addMessage }) => {
  const { dark, toggleDark } = useContext(ThemeContext);

  const handleOnClick = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    if (toggleDark) {
      toggleDark();
    }
  };

  const [text, setText] = useState('');

  const handleText = (ev: React.FormEvent<HTMLFormElement>) => {
    ev.preventDefault();
    addMessage({
      text,
      author: 'User',
    });
    setText('');
  };

  return (
    <>
      <form onSubmit={handleText}>
        <Input value={text} onChange={(ev) => setText(ev.target.value)} />

        <Button variant="contained" type="submit">
          Send
        </Button>
      </form>

      <h1>{dark ? 'ðŸŒ™' : 'ðŸŒž'}</h1>
      <button onClick={handleOnClick}>Toggle dark mode</button>
    </>
  );
};
