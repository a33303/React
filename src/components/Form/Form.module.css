.sendMSg {
    width: 34.5%;
}