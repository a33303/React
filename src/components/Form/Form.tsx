import React, { useState, FC } from 'react';
import { Button, TextField } from '@mui/material';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { AUTHORS } from '../../constants';
import { set } from 'firebase/database';
import { getMessagesListRefId } from '../../services/firebase';
import { nanoid } from 'nanoid';
import styles from "../Form/Form.module.css";

export const Form: FC = () => {
  const dispatch = useDispatch();
  const { chatId } = useParams<{ chatId?: string }>();
  const [text, setText] = useState('');

  const handleSubmit = (ev: React.FormEvent<HTMLFormElement>) => {
    ev.preventDefault();
    if (chatId) {
      const id = nanoid();
      set(getMessagesListRefId(chatId, id), {
        id,
        text,
        author: AUTHORS.user,
      });
      // dispatch(
      //   addMessageWithThunk({
      //     chatId,
      //     text,
      //     author: AUTHORS.user,
      //   })
      // );
    }

    setText('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <TextField
          label="Enter message"
          value={text}
          className={styles.sendMSg}
          onChange={(ev) => setText(ev.target.value)}
      />
      <Button type="submit">
        Send
      </Button>
    </form>
  );
};
