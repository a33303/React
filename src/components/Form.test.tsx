import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Form } from './Form';

describe('From', () => {
  it('renders From component', () => {
    render(<Form addMessage={() => null} />);
    expect(screen.getByText('Send')).toBeInTheDocument();
  });
  it('test snapshot', () => {
    const { asFragment } = render(<Form addMessage={() => null} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
