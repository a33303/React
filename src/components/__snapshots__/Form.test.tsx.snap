// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From test snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-a88p61-MuiInputBase-root-MuiInput-root"
    >
      <input
        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="submit"
    >
      Send
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
  <h1>
    ðŸŒ™
  </h1>
  <button>
    Toggle dark mode
  </button>
</DocumentFragment>
`;
